### PR TITLE
Possibility to restore original value of the field when upload failed.

### PR DESCRIPTION
**Pull request by Michael-Gusev**
It helps to keep original value if we just have file field on form and don't want to upload it but keep original file.

@josegonzalez, could you review?

### CODE REVIEW

File: docs/configuration.rst [hunk 1]
```diff
@@ -65,3 +65,7 @@ passed in under each field in your behavior configuration.
 -  ``keepFilesOnDelete``: Keep *all* files when deleting a record.
 
    -  Default: (boolean) ``true``
+
+-  ``restoreValueOnFailure``: Restores original value of the current field when uploaded file has error
+
+   - Defaults: (boolean) ``false``
```
Comment (ADmad) [inline docs/configuration.rst]: Defaulting to `true` would make more sense imo. No one want to lose existing data :)

**Comment by josegonzalez** [comment]
@Michael-Gusev can you adjust to @ADmad's suggestion? Sane defaults are great :)

**Comment by Michael-Gusev** [comment]
@josegonzalez @ADmad, code has been updated.

**Comment by josegonzalez** [comment]
Looks good to me, thoughts @ADmad before I merge this?

**Comment by ADmad** [comment]
LGTM but even with current code I don't see how your existing values would be overwritten. The `beforeMarshal()` ensures that field is unset from data when no file is uploaded.

@Michael-Gusev Have you actually experienced existing values being lost when editing record without uploading a file?

**Comment by Michael-Gusev** [comment]
@ADmad, I have simple form.
```php
echo $this->Form-input('image', ['type' => 'file'])
```
in controller
```php
$this->Products->patchEntity($entity, $this-request->data);
$this->Products->save($entity);
```
so when I save form without choosing file it sets value of image to null.

**Comment by ADmad** [comment]
I am unable to reproduce the behavior you mention. Though I do get an error

```
Notice (8): Array to string conversion [CORE/src/Database/Statement/PDOStatement.php, line 72]
```

when an upload error like `UPLOAD_ERR_INI_SIZE` is set for the field and validation rules are not set to check for upload errors. So restoring original value seems reasonable to me.

Personally  after restoring original value I would also set the field as not dirty so as to avoid unnecessarily re-saving existing value.

**Comment by Michael-Gusev** [comment]
@ADmad, you're right, also updated dirty.

**Pull request merged**